File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode3136ValidWord.cpp
#include "LeetCode3136ValidWord.h"
using namespace std;
bool isLetter(char c)
{
    if (c>='A' && c<='Z')
        return true;
    if (c>='a' && c<='z')
        return true;
    return false;
}
bool isValidChar(char c)
{
    if (c>='0' && c<='9')
        return true;
    if (isLetter(c))
        return true;
    return false;
}
bool isVowel(char c)
{
    if (c=='a' || c=='e' || c=='i' || c=='o' || c=='u')
        return true;
    if (c=='A' || c=='E' || c=='I' || c=='O' || c=='U')
        return true;
    return false;
}
bool isConsonant(char c)
{
    if (isVowel(c))
        return false;
    if (isLetter(c))
        return true;
    return false;
}
bool LeetCode3136ValidWord::isValid(string word)
{
    int length = word.length();
    if (length<3)
        return false;
    int vowelCount{};
    int consonantCount{};
    for (auto c:word)
    {
        if (!isValidChar(c))
            return false;
        if (isConsonant(c))
            consonantCount++;
        if (isVowel(c))
            vowelCount++;
    }
    return vowelCount>0&&consonantCount>0;
}

File: LeetCodeProblemsGo/LeetCode3136ValidWord/LeetCode3136ValidWord.go
package LeetCode3136ValidWord

func isValid(word string) bool {
	length := len(word)
	if length < 3 {
		return false
	}
	vowelCount := 0
	consonantCount := 0
	for _, char := range word {
		if !isValidCharacter(char) {
			return false
		}
		if isVowel(char) {
			vowelCount++
		}
		if isConsonant(char) {
			consonantCount++
		}
	}
	if vowelCount < 1 || consonantCount < 1 {
		return false
	}
	return true
}

func isLetter(b rune) bool {
	if b >= 'a' && b <= 'z' {
		return true
	}
	if b >= 'A' && b <= 'Z' {
		return true
	}
	return false
}
func isVowel(b rune) bool {
	if b == 'a' || b == 'e' || b == 'i' || b == 'u' || b == 'o' {
		return true
	}
	if b == 'A' || b == 'E' || b == 'I' || b == 'U' || b == 'O' {
		return true
	}
	return false
}
func isConsonant(b rune) bool {
	if isVowel(b) {
		return false
	}
	if !isLetter(b) {
		return false
	}
	return true
}
func isValidCharacter(b rune) bool {
	if isLetter(b) {
		return true
	}
	if b >= '0' && b <= '9' {
		return true
	}
	return false
}
